more_itertools/recipes.py

```diff
@@ -304,26 +304,42 @@ def powerset(iterable):
 
 
 def unique_everseen(iterable, key=None):
-    """Yield unique elements, preserving order.
```

Comment:
Yes, this is in the original set of recipes from the Python docs. @abarnert's changes improve it to work with unhashable items, as `unique_justseen` does.